src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -54,13 +57,14 @@ export default class GraphFilter extends React.PureComponent<GraphFilterProps> {
     super(props);
   }
 
-  updateDuration = (value: number) => {
-    if (this.props.graphDuration.value !== value) {
-      this.props.onDurationChange({ value: value });
+  handleDuration = (duration: number) => {
+    if (this.props.graphDuration.value !== duration) {
+      this.props.setDuration({ value: duration }); // inform Redux of duration change
+      this.props.onDurationChange({ value: duration });
```

Comment:
I'm wondering if there is a way to unify these callbacks. It looks redundant to have both since are very related/coupled.

Reply:
With Redux, probably the parent callbacks can all be eliminated. I just wasn't sure about making those changes yet.

Reply:
+1, it can probably follow the approach now in place for activeNamespace.  Avoid the drill-down handlers and use redux to set the value, and react to the change in the appropriate containers, like GraphPage.

Reply:
Removing the drill-down onDuration() should probably be done as an immediate follow-up task to this PR because it really shouldn't be necessary if the redux state change is properly handled by relevant containers.